### ISSUE TITLE
How to connect ESP8266

### ISSUE THREAD

**Issue by Mark-81** [marvinroger/async-mqtt-client#85]
Thanks for your work! I have a simple question: how to connect a ESP8266 module to the Arduino? I.e. I have an Olimex one. I need to connect it to the main Arduino UART? I don't see any configuration in the code.

**Comment by tonton81** [comment]
why you need to connect it to arduino? you want the arduino as a slave?

**Comment by Mark-81** [comment]
Because right now it's the only way I know... I always used an AVR as a "master" that sends AT commands to the ESP8266 module. Without this connection how I can use the ESP8266 module?

**Comment by tonton81** [comment]
you program arduino ONTO the esp8266 same as arduino

**Comment by Mark-81** [comment]
Got it, thanks! Last question - I apologize. That means I can use any module if I would able to replace the firmware or just place the ESP8266 on my own PCB, right?

**Comment by tonton81** [comment]
module i presume you mean library? yeah alot of arduino libraries can run on esp8266 just like arduino does, the arduino IDE flashes the ESP with your sketch and itll work just like an arduino after

**Comment by Mark-81** [comment]
No, I'm talking about physical modules, like this: https://www.olimex.com/Products/IoT/MOD-WIFI-ESP8266-DEV/open-source-hardware.

I designed a complex board with an xmega that does a lot of stuff. Then I need to provide WiFi connection. Right now I have the module above connected using the UART and I send AT commands to connect to an AP, open a socket and send some data.

To use this library instead I need to burn a custom firmware on the Olimex module and adapt the example code in order to receive data (i.e. sensors) from the xmega and publish them to the broker.

**Comment by tonton81** [comment]
the olimex is nothing but an esp8266 module flashed with AT command set. you could drop all your xmega code on the esp8266 (olimex) and run it without the xmega, if you have other connections on the xmega you can bring them over but be aware some ESP pins are special and can’t be used without caution

**Comment by tonton81** [comment]
the ESP also has more ram flash and storage and faster CPU than your xmega, which the AT command set doesnt put to good use :)
